.github/workflows/commit-format.yml

```diff
@@ -41,4 +52,32 @@ jobs:
             fi
           done
           exit $retval
-     
+
+  check_commit_signoff:
+    runs-on: ubuntu-latest
+    name: check-commit-signoff
+    
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: 0
+
+      - name: Check for `Signed-off-by` line
+        run: |
+          if echo ${{ secrets.FSF_CONTRIBUTORS }} | grep ${{ github.event.pull_request.user.login }}; then
+            echo "author has an FSF copyright assignment"
+          fi
+
+          retval=0
+          for commit in $(git rev-list origin/${{ github.event.pull_request.base.ref }}..${{ github.event.pull_request.head.sha }});
+          do
+            echo -n "Checking commit sign-off for $commit: "
+            if [[ $(git log $commit -1 --format="%B" | tail -n 2) = Signed-off-by:* ]]; then
```

Comment:
Need not just any `Signed-off-by:*`, but rather that it does match the Git commit Author or `Co-authored-by` tag(s)?

Reply:
Right. Let me know if the script sort of looks good to you in its current form and I'll make that change :)